rest/src/main/java/fi/otavanopisto/pyramus/rest/StudentRESTService.java

```diff
@@ -1760,8 +1769,33 @@ public Response listStudentContactLogEntriesByStudent(@PathParam("STUDENTID") Lo
     Student student = studentController.findStudentById(studentId);
     Status studentStatus = checkStudent(student);
     if (studentStatus != Status.OK)
-      return Response.status(studentStatus).build();
-
+      return Response.status(studentStatus).build(); 
+    
+    User loggedUser = sessionController.getUser();
+    StaffMember staffMember = userController.findStaffMemberById(loggedUser.getId());
```

Comment:
Should first check if `loggedUser` is `ADMINISTRATOR`. If so, further access checks aren't needed and all entries (and their comments) can be returned.

If not, check that `loggedUser` is not a `STUDENT` (can be checked simply by checking if `staffMember` is `null`). If they are, kick them out.

Next, figure out whether `loggedUser` (now certainly a staff member at least) is the guidance councelor of the student. If so, return all entries (and their comments). If not, only return own entries and comments.

Also see further review comments to simplify this.